setup.py

```diff
@@ -5,15 +5,19 @@
 from setuptools.extension import Extension
 
 SANITIZE = os.environ.get("MSGSPEC_SANITIZE", False)
-DEBUG = os.environ.get("MSGSPEC_DEBUG", SANITIZE)
+COVERAGE = os.environ.get("MSGSPEC_COVERAGE", False)
+DEBUG = os.environ.get("MSGSPEC_DEBUG", SANITIZE or COVERAGE)
 
 extra_compile_args = []
 extra_link_args = []
-if DEBUG:
-    extra_compile_args.extend(["-O0", "-g", "-UNDEBUG"])
 if SANITIZE:
     extra_compile_args.extend(["-fsanitize=address", "-fsanitize=undefined"])
     extra_link_args.extend(["-lasan", "-lubsan"])
+if COVERAGE:
+    extra_compile_args.append("--coverage")
+    extra_link_args.append("-lgcov")
```

Comment:
Here we add compile and link args to enable `gcov` support. These could have been added in CI only using `CFLAGS` and `LDFLAGS` respectively, but I find adding a top-level option in `setup.py` easier for me to use locally during development.

This lets you build the package with gcov support using:

```
MSGSPEC_COVERAGE=true python setup.py develop
```